frontend/src/components/common/table/useSortProperty.tsx

```diff
@@ -0,0 +1,16 @@
+import { useState } from "react";
+
+const useSortProperty = <SortPropertyArrayType extends readonly string[]>(
+  initialSortProperty:
+    | SortPropertyArrayType[number]
+    | (() => SortPropertyArrayType[number]),
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  sortPropertyArray?: SortPropertyArrayType,
```

Comment:
?

Reply:
This makes calling the hook slightly cleaner. So rather than passing `useSortProperty<type of PROPS>("default")` you can do `useSortProperty("default", PROPS);` It does mean that we have an unused variable though, but I thought it made the code clearer overall. Thoughts?